publish.js

```diff
@@ -88,7 +88,7 @@ function patchedManifest (spec, auth, base, opts) {
     // currently gets filled in by the npm registry itself, based on auth
     // information.
     manifest._npmUser = {
-      username: auth.username,
+      name: auth.username,
```

Comment:
I think the underlying issue is that the conditional that wraps the `_npmUser` assignment isn't _also_ checking for `auth._auth`, and if found, splitting the `<username>:<password>` field appropriately. I can't speak to the correctness of the "username" vs "name" property itself (although I suppose it should be what `normalize-package-data` [does](https://github.com/npm/normalize-package-data/blob/df8ea05b8cd38531e8b70ac7906f420344f55bab/lib/fixer.js#L349-L369), so yes, `name` is the correct field AFAICT).

```js
if (auth.username || auth.email || auth._auth) {
  manifest._npmUser = {
    name: auth.username || (auth._auth && auth._auth.split(':').shift()),
    email: auth.email,
  }
}
```

Reply:
I don't know if `username` if recognized by the registry, but I can say for sure that `name` is because it's the way previous version of npm worked.

Reply:
thanks for looking into this y'all. I'm working right this second on adding proper _auth support to `npm-registry-fetch`, where I think this should live.